File: testsuite/results/operatorTemplate.nim
## ! @relates cv::Vec
## ! @{

proc operatorPE*[_Tp1; _Tp2; cn: static[cint]](a: var Vec[_Tp1, cn]; b: Vec[_Tp2, cn]): var Vec[
    _Tp1, cn] =
  discard

proc `+=`*[_Tp1; _Tp2; cn: static[cint]](a: var Vec[_Tp1, cn]; b: Vec[_Tp2, cn])

File: testsuite/tests/operatorTemplate.cpp
//! @relates cv::Vec
//! @{

template<typename _Tp1, typename _Tp2, int cn> static inline
Vec<_Tp1, cn>& operatorPE (Vec<_Tp1, cn>& a, const Vec<_Tp2, cn>& b)
{
    for( int i = 0; i < cn; i++ )
        a.val[i] = saturate_cast<_Tp1>(a.val[i] + b.val[i]);
    return a;
}

template<typename _Tp1, typename _Tp2, int cn> static inline
Vec<_Tp1, cn>& operator+= (Vec<_Tp1, cn>& a, const Vec<_Tp2, cn>& b)
{
    for( int i = 0; i < cn; i++ )
        a.val[i] = saturate_cast<_Tp1>(a.val[i] + b.val[i]);
    return a;
}